terraform/azure/modules/sqlmi/providers.tf

```diff
@@ -0,0 +1,14 @@
+terraform {
+  required_version = ">=1.0"
+
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~>3.0"
+    }
+  }
+}
+
+provider "azurerm" {
```

Comment:
line12, provider needs to be set in the calling module (the root of the deployment module). It cannot be here or we will face issue during deployment